platform_versioned_docs/version-24.1/data_studios/custom-envs.mdx

```diff
@@ -0,0 +1,143 @@
+
+## Conda packages
+
+### Prerequisites
+
+- Wave must be configured. For more information, see [Wave containers][wave].
+
+### Conda package syntax {#conda-package-syntax}
+
+When you add a new data studio, you can customize its configuration to install a list of Conda packages to the template. The supported schema is identical to that used by the Conda `environment.yml` environment file. For more information on the Conda environment file, see [Creating an environment file manually][env-manually].
+
+```yaml title="Example environment.yml file"
+name: example-env
+channels:
+  - defaults
+  - conda-forge
+dependencies:
+  - python=3.7
+  - numpy
+  - pandas
+  - scikit-learn
+  - pip:
+    - matplotlib
+    - seaborn
+    - jupyter
+```
+
+To create a data studio with custom Conda packages, see [Add a data studio][add-ds].
+
+## Custom container template image {#custom-containers}
+
+For advanced use cases, you can build your own container template image.
+
+:::note
+Amazon Elastic Container Registry (ECR) is the only supported private container registry.
+:::
+
+### Prerequisites
+
+- Access to a container image repository, either a public container registry or in a private Amazon ECR repository
+- A container template image
+
+### Dockerfile configuration {#dockerfile}
+
+For your custom template container image, you must use a Seqera provided base image and include several additional build steps for compatibility with Data Studios.
+
+#### Ports
+
+The container must use the value of the `CONNECT_TOOL_PORT` environment variable as the listening port for whatever interactive software that you include in your custom container.
+
+#### Signals
+
+Upon termination, the container's main process must handle the `SIGTERM` signal and perform any necessarily cleanup. After a 30 second grace period, the container receives the `SIGKILL` signal.
+
+#### Minimal Dockerfile
+
+The minimal Dockerfile includes directives to accomplish the following:
+
+- Pulls a Seqera-provided base image with prerequisite binaries
+- Installs prerequisite packages, such as wget and fuse
+- Installs and configures Seqera Fusion
+- Sets the container entry points
+
+Customize the following Dockerfile to include any additional software that you require:
+
+```docker title="Minimal Dockerfile"
+# Seqera base image
+FROM public.cr.seqera.io/platform/connect-client:0.7.1 as connect
+
+# 1. Add connect binary
+COPY --from=connect /connect-client /usr/bin/connect-client
+
+# 2. Install required system packages
+RUN apt-get update --yes && apt-get install --yes --no-install-recommends \
+    ca-certificates \
+    wget \
+    fuse \
+    btrfs-progs \
+    squashfs-tools \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# 3. Install Fusion
```

Comment:
Really dumb question: why don't we just provide a base image with Fusion pre-baked, the connect client in the right place etc, so users don't have to do all this every time?

Reply:
I think this is the only outstanding issue to be resolved.